src/MultiTypeSymbolTable.chpl

```diff
@@ -470,6 +471,27 @@ module MultiTypeSymbolTable
                 return false;
             }
         }
+
+        /*
+        Attempts to find all sym entries that match the provided regex string, then 
+        returns a string array of matching names
+
+        :arg pattern: regex string to search for
+        :type pattern: string
+
+        :returns: string array containing matching entry names
+        */
+        proc findAll(pattern: string): [] string throws {
+            var regex = compile(pattern);
+            var infoStr = "";
+            for name in tab {
+                if name.match(regex) {
+                    infoStr += name + "+";
+                }
+            }
+            infoStr = infoStr.strip("+");
+            return infoStr.split("+");
+        }
```

Comment:
using a [Reduce Intent](https://chapel-lang.org/docs/technotes/reduceIntents.html#reduce-intents), you can implement this in parallel (provided you don't care about order)
```Chapel
proc findAll(pattern: string): [] string throws {
    var regex = compile(pattern);
    var infoStr = "";
    forall name in tab with (+ reduce infoStr) {
        if name.match(regex) {
            infoStr += name + "+";
        }
    }
    return infoStr.strip("+").split("+");
}
```

Reply:
I'm curious if @ronawho has any suggestions here

Reply:
I suspect it doesn't really matter here, and I'd go with whatever code is cleanest. My guess (though I don't really know) is that there's probably not more than a couple dozen or hundred sym entries at any given time.

If you have "enough" entries a parallel reduce will definitely be faster, but for a small reduce the array creation and task creation overheads will outweigh any parallelism benefits. That said, the overhead of those is small so even if things are slower, it's imperceptibly slower to the Arkouda user waiting so I maybe I'd go with the parallel approach just in case there are ever a lot of entries.

And note that this doesn't get more efficient when running on multiple nodes. `keysToArray` is returning a local non-distributed array and operations on it will still be local. (unless you have like a million or more entries, I don't think you want to distribute this so this is the correct behavior)

Reply:
Awesome! Thanks ronawho